sklearn/metrics/pairwise.py

```diff
@@ -78,6 +78,10 @@ def check_pairwise_arrays(X, Y, precomputed=False):
         True if X is to be treated as precomputed distances to the samples in
         Y.
 
+    dtype : string, type, list of types or None (default=None)
```

Comment:
I took it from [check_array](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/validation.py#L292), and I tend to prefer it this way.
I feel what you suggest is somehow a recursive definition, and I don't really know if it's less confusing.